lib/OpenQA/Constants.pm

```diff
@@ -107,6 +108,8 @@ use constant DB_TIMESTAMP_ACCURACY => 1;
 use constant VIDEO_FILE_NAME_START => 'video.';
 use constant VIDEO_FILE_NAME_REGEX => qr/^.*\/video\.[^\/]*$/;
 
+use constant FRAGMENT_REGEX => qr'(#([-?/:@._~!$&\'()*+,;=\w]|%[0-9a-fA-F]{2})*)*';
```

Comment:
please use https://perldoc.perl.org/perlre#Character-Classes-and-other-Special-Escapes

Reply:
Thanks, I fixed it a bit to remove `_` since it is contained in the `\w` sequence. For the last part, I don't want to match `_` so I refrained from using `\w`, explicitly stating letters and numbers.

I wonder: based on the definition of fragment identifiers, `#` is not in the allowed characters for a fragment (apart from the obvious, at the start). I have read that some browsers are allowing it, though. In the current implementation, `#` is allowed in the fragment identifier body. Should I leave it like this?

Reply:
leave it